emission/tests/modellingTests/TestGreedySimilarityBinning.py

```diff
@@ -10,44 +11,111 @@ def setUp(self) -> None:
+                model_config = {
+                                    "metric": "od_similarity",
+                                    "similarity_threshold_meters": binning_threshold,  # meters,
+                                    "apply_cutoff": False,
+                                    "clustering_way": cw,  
+                                    "incremental_evaluation": False
+                                }
+                model= eamtg.GreedySimilarityBinning(model_config)
+                model.fit(trips)
+                #check each bins for no of trips
+                no_large_bin = all(map(lambda b: len(b['feature_rows']) == 1, model.bins.values()))
+                #Since all trips were sampled outside the threshold, there should be no bin
+                # with more then 1 trip
+                self.assertTrue(no_large_bin,"no bin should have more than 1 features in it")
 
-        # generate $n trips. $m of them should have origin and destinations sampled
+    def testBinning(self):
+        """
+        Tests the three (origin, destination and origin-destination based) 
+        binning configuration for trips.
+
+        When the points lie within threshold ,the trips are binned together.
+        """
+        # generate $n trips. $m of them should have origin sampled
         # within a radius that should have them binned.
         n = 20
         m = 5
-        trips = etmm.generate_mock_trips(
-            user_id="joe", 
-            trips=n, 
-            origin=(0, 0), 
-            destination=(1, 1), 
-            label_data=label_data, 
-            within_threshold=m, 
-            threshold=0.001,  # ~ 111 meters in degrees WGS84
-        )
-
-        # pass in a test configuration to the binning algorithm
-        model_config = {
-            "metric": "od_similarity",
-            "similarity_threshold_meters": 500,  # meters,
-            "apply_cutoff": False,
-            "incremental_evaluation": False
+        binning_threshold=500
+        label_data = {
+            "mode_confirm": ['walk', 'bike', 'transit'],
+            "purpose_confirm": ['work', 'home', 'school'],
+            "replaced_mode": ['drive']
         }
-        model = eamtg.GreedySimilarityBinning(model_config)
-        
-        model.fit(trips)
 
-        # $m trip features should appear together in one bin
-        at_least_one_large_bin = any(map(lambda b: len(b['feature_rows']) == m, model.bins.values()))
-        self.assertTrue(at_least_one_large_bin, "at least one bin should have at least 5 features in it")
+        # parameters passed for testing. A list, where each element of this list takes the form 
+        # [trip part to be sampled within mentioned threshold , clustering way used to check similarity]
+        parameters= [["o_",'origin'],["_d",'destination'],["od",'origin-destination']]
+        for tp,cw in parameters:
+            with self.subTest(trip_part=tp,clustering_way=cw):
```

Comment:
I think that subTest largely includes what I had in mind for avoiding the duplication of `model_config` creation code.

Reply:
yes it does. That's how  I was able to reduce a lot of duplication.